### PR TITLE
redis: Support Sentinel with SSL

### PR DESCRIPTION
**Pull request by auvipy**
Use the SentinelManagedSSLConnection when SSL is enabled for the
transport.

### CODE REVIEW

**Review by matusvalo — CHANGES_REQUESTED** [review]
The unittests are failing.

**Comment by AbdealiJK** [comment]
@auvipy Just saw that you created this - Thanks :)
I've modified the PR to create the `SentinelManagedSSLConnection` class directly in Kombu for now - until https://github.com/andymccurdy/redis-py/pull/1419 gets merged.

This way @matusvalo - the unit tests will pass and we can switch to the redis-py class when that gets merged

This PR relates to: https://github.com/celery/celery/issues/6455

**Review by auvipy — COMMENTED** [review]
now just flake8 n doc styles are failing

**Comment by AbdealiJK** [comment]
Thanks for the bump :) Fixed

**Comment by matusvalo** [comment]
Thanks for PR. It seems fine. Could you add also unittests?

**Comment by AbdealiJK** [comment]
@matusvalo - Added unit test.
For the integration test - Yeah, I'm not sure where to start on that

**Comment by AbdealiJK** [comment]
Seems like the test failure is now due to azure-servicebus releasing a new version 7 hours ago :(
https://pypi.org/project/azure-servicebus/7.0.0/

**Review by matusvalo — COMMENTED** [review]
Thank you @AbdealiJK. I have just one comment regarding unittests.

About Integration test, see the following comment: https://github.com/celery/kombu/issues/1254#issuecomment-735296395

File: t/unit/transport/test_redis.py [hunk 1]
```diff
@@ -1465,3 +1465,22 @@ def test_missing_master_name_transport_option(self):
...
+            with Connection(
+                    'sentinel://',
+                    transport_options={'master_name': 'not_important'},
+                    ssl=ssl_params) as conn:
+                params = conn.default_channel._connparams()
+                assert params['ssl_cert_reqs'] == ssl_params['ssl_cert_reqs']
+                assert params['ssl_ca_certs'] == ssl_params['ssl_ca_certs']
+                assert params['ssl_certfile'] == ssl_params['ssl_certfile']
+                assert params['ssl_keyfile'] == ssl_params['ssl_keyfile']
+                assert params.get('ssl') is None
```
[the remark this answers is not in the document]
Comment (AbdealiJK) [reply t/unit/transport/test_redis.py:1486]: Fixed

**Pull request merged**